docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,292 @@
+       - **No Impact on Existing Setup:** Avoids modifying the ETCD cluster's replicas, preventing any unintended consequences.
+       - **Automated Process:** Can be integrated into existing maintenance jobs without manual intervention.
+
+**Proposed Solution:**
+
+*Option B is recommended* because it respects the operator's intention to keep the ETCD cluster hibernated and avoids the complexities and risks associated with modifying the cluster's state.
+
+Leverage the compaction job to take fresh snapshots periodically during hibernation. This ensures that:
+
+- **Immutability is Maintained:** New snapshots have a fresh immutability period, keeping backups protected.
+- **Operator Intentions are Respected:** The ETCD cluster remains in its hibernated state, as per the operator's intention.
+- **Storage Costs are Controlled:** Avoids unnecessary extension of immutability for all snapshots, preventing increased storage costs.
+
+### New Compaction Controller Flag
+
+**Flag:** `--hibernation-snapshot-interval`
+
+- **Type:** Duration
+- **Default:** `24h`
+- **Description:** This flag sets the period after which a compaction job should be triggered for a hibernated ETCD cluster, based on the time since the last renewal of the full snapshot lease. If the time since `fullLease.Spec.RenewTime.Time` exceeds the duration specified by this flag, and `etcd.spec.replicas` is `0` (indicating hibernation), the compaction job will automatically trigger to create a new snapshot. This approach ensures that backups remain within the immutability period and are safeguarded against becoming mutable.
+
+### Garbage Collection during Compaction for Hibernated ETCD Clusters
+
+If an ETCD cluster is hibernated for a long duration, there is a chance that backup storage will accumulate a large number of snapshots. Since the ETCD cluster is not running, the `etcd-backup-restore` component is also not running to perform garbage collection of the snapshots.
+
+To address this, the compaction job should be enhanced to handle garbage collection during hibernation. This ensures that old snapshots are cleaned up appropriately, considering the immutability constraints.
+
+### Excluding Snapshots Under Specific Circumstances
+
+Given that immutable backups cannot be deleted until the immutability period expires, there are scenarios, such as corrupted snapshots or other anomalies, where certain snapshots must be skipped during the restoration process. To facilitate this:
+
+- **Custom Metadata Tags:** Utilize custom metadata to mark specific objects (snapshots) that should be bypassed. To exclude a snapshot from the restoration process, attach custom metadata to it with the key `x-etcd-snapshot-exclude` and value `true`. This method is officially supported, as demonstrated in the [etcd-backup-restore PR](https://github.com/gardener/etcd-backup-restore/pull/776).
```

Comment:
Its not clear from the doc who takes care of attaching the custom metadata flag and how its consumed? Can we describe here to avoid any unintended interpretation of the flow.

Reply:
Human operators add these tags; will include this.